src/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs

```diff
@@ -20,8 +20,68 @@ namespace System.Net.Security.Tests
 {
     using Configuration = System.Net.Test.Common.Configuration;
 
+#if TargetsLinux
```

Comment:
We shouldn't be adding this kind of logic with ifdefs here.  This kind of stuff should move to Common with the rest of the PlatformDetection stuff.

Reply:
Why not, the openssl version detection doesn't have any other use case other than system.net.security library so far. Makes sense to localize the change to just here. The libcurl version checks are also localized to System.Net.Http.Funtional.Tests with the #ifdef pattern, this is keeping in line with the existing pattern.

Reply:
I agree with David.  You're adding CryptoNative_SSLEayVersion to System.Security.Cryptography.Native; and as with other interop for accessing such functions, the P/Invokes for working with those should be defined in the src\Common\Interop folder.

Reply:
ok, will put it in common, since most folks voted that.